Switch to paho-mqtt 2.x

mmatuska opened the pull request:
Keep using deprecated V1 callback API for now

mmatuska committed:
Switch to paho-mqtt 2.x

Keep using deprecated V1 callback API for now

jbouwh commented on requirements-dev.txt:
```diff
@@ -1,4 +1,4 @@
-paho-mqtt>=1.5.0,<2.0.0
+paho-mqtt>=2.0.0,<3.0.0
```
```suggestion
paho-mqtt>=2.1.0,<3.0.0
```

This will fix the compatibility issues with HA Core 2025.3

jbouwh reviewed (approved):
LGTM,
Thanks @mmatuska 👍

jbouwh commented:
@albertogeniola Can you have a look here?

JosephRDawson commented:
> @albertogeniola Can you have a look here?

Sadly I am not a HA developer but I think we may need someone to fork this project.  Not sure if the main developer has moved on or not. But he hasn't responded to any comments in the last few weeks. Hopefully he is still well and on vacation.

jbouwh commented:
> > @albertogeniola Can you have a look here?
> 
> Sadly I am not a HA developer but I think we may need someone to fork this project. Not sure if the main developer has moved on or not. But he hasn't responded to any comments in the last few weeks. Hopefully he is still well and on vacation.

Last merge was 3 weeks ago,  so it seems to me we should just be patient. HA users could downgrade to the latest stable version IMO

mmatuska commented:
I have updated this PR against latest code and pushed the paho-mqtt requirement to 2.1.0
Also changed base branch to development-0.4.X.X, as the project owner seems to push only releases to 0.4.X.X

a8 commented:
Thank you @mmatuska! Updating to paho-mqtt 2.1.0 did the trick and works.

albertogeniola commented:
TY @mmatuska for the PR. Looks good to me. I'll now release a new hotfix for HA folks.
